component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java

```diff
@@ -158,7 +159,11 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
             throws AuthenticationFailedException {
 
         String username = null;
-
+        String showAuthFailureReasonOnLoginPage = null;
+        Map<String, String> parameterMap = getAuthenticatorConfig().getParameterMap();
+        if (parameterMap != null) {
+            showAuthFailureReasonOnLoginPage = parameterMap.get(TOTPAuthenticatorConstants.CONF_SHOW_AUTH_FAILURE_REASON_ON_LOGIN_PAGE);
+        }
```

Comment:
can move this logic to https://github.com/wso2-extensions/identity-outbound-auth-totp/pull/135/files#diff-8d5c29187e52b2268d09add06e3537a10e090368a0b9e48583079ea098a5eea0R213

Reply:
This logic is refactored in 917a93c15aa7fb493b77d849f62205caa2623e89